### PR TITLE
invalidate cache

### LINKED ISSUE #167: Add button to invalidate cache/force fetch data from marketplace
The json data as queried from the marketplace is cached for 24hrs.

We shall give the admin the opportunity to force refresh.

Code location where data is written to the cache: https://github.com/owncloud/market/blob/master/lib/MarketService.php#L517

### PR DESCRIPTION
**Pull request by DeepDiver1975**
fixes #167

### CODE REVIEW

File: src/components/ApiForm.vue [hunk 1]
```diff
@@ -39,6 +41,9 @@
 			openModalViewKey () {
 				UIkit.modal('#view-api-key').toggle();
 			},
+			forceAppDataReload() {
```
Comment (DeepDiver1975) [inline src/components/ApiForm.vue]: yes - I suck at UI stuff .....

Comment (felixheidecke) [reply src/components/ApiForm.vue]: I'll take over from here ;-)

Comment (felixheidecke) [reply src/components/ApiForm.vue]: I'll take over from here :-)

File: lib/Controller/MarketController.php [hunk 2]
```diff
@@ -303,4 +303,15 @@ public function requestDemoLicenseKeyFromMarket() {
 			);
 		}
 	}
+
+	public function invalidateCache() {
```
Comment (felixheidecke) [inline lib/Controller/MarketController.php]: No exception catching here?

File: lib/Controller/MarketController.php [hunk 3]
```diff
@@ -303,4 +303,15 @@ public function requestDemoLicenseKeyFromMarket() {
...
+		$this->marketService->invalidateCache();
+		return new DataResponse(
+			[
+				'message' => $this->l10n->t('Cache invalidate. Reload to grab the latest apps from the marketplace.')
```
Comment (felixheidecke) [inline lib/Controller/MarketController.php]: There is an automatic reload after invalidating the cache. Please change msg accordingly. (Eg. "Chache cleared")

Comment (DeepDiver1975) [reply lib/Controller/MarketController.php]:
```sql
GRANT PERMISSION PHP_CODE_CHANGE  to @felixheidecke

File: lib/Controller/MarketController.php (hunk 2 above)
Comment (DeepDiver1975) [reply lib/Controller/MarketController.php]: No exceptions are expected

**Comment by DeepDiver1975** [comment]
@felixheidecke I tested this and it works as far as I can tell - THX for the UI work

**Pull request merged**